v0/destinations/google_ads_enhanced_conversion/transform.js

```diff
@@ -0,0 +1,119 @@
+const {
+  getSuccessRespEvents,
+  getErrorRespEvents,
+  CustomError,
+  constructPayload,
+  defaultRequestConfig,
+  getValueFromMessage
+} = require("../../util");
+
+const { trackMapping, BASE_ENDPOINT } = require("./config");
+
+const responseBuilder = async (metadata, message, { Config }, payload) => {
+  const response = defaultRequestConfig();
+  const { event } = message;
+  let flag = 0;
+  const { listOfConversions } = Config;
+  for (let i = 0; i < listOfConversions.length; i += 1) {
+    if (listOfConversions[i].conversions === event) {
+      flag = 1;
+      break;
+    }
+  }
+  if (event === undefined || event === "" || flag === 0) {
+    throw new CustomError(
+      `[Google_ads_enhanced_marketing]:: Conversion named ${event} is not exist in rudderstack dashboard`,
+      400
+    );
+  }
+
+  response.endpoint = `${BASE_ENDPOINT}/${Config.customerId}:uploadConversionAdjustments`;
+  response.body.JSON = payload;
+  const accessToken = metadata.secret.access_token;
+  response.headers = {
+    Authorization: `Bearer ${accessToken}`,
+    "Content-Type": "application/json",
+    "developer-token": getValueFromMessage(metadata, "secret.developer_token")
+  };
+  response.params = { event, customerId: Config.customerId };
+  if (Config.subAccount)
+    if (Config.loginCustomerId)
+      response.headers["login-customer-id"] = Config.loginCustomerId;
+    else
+      throw new CustomError(
+        `[Google_ads_enhanced_marketing]:: loginCustomerId is required as subAccount is true.`,
+        400
+      );
+  return response;
+};
+
+const processTrackEvent = async (metadata, message, destination) => {
+  const payload = constructPayload(message, trackMapping);
+  if (!payload.conversionAdjustments[0].userIdentifiers) {
+    throw new CustomError(
+      `[Google_ads_enhanced_marketing]:: Any of email, phone, firstName, lastName, city, street, countryCode, postalCode or streetAddress is required in traits.`,
+      400
+    );
+  }
+  if (typeof payload.partialFailure !== "boolean") {
+    throw new CustomError(
+      `[Google_ads_enhanced_marketing]:: partialFailure should be a boolean.`,
+      400
+    );
+  }
+  payload.conversionAdjustments[0].adjustmentType = "ENHANCEMENT";
+  // Removing the null values from userIdentifier
+  const arr = payload.conversionAdjustments[0].userIdentifiers;
+  payload.conversionAdjustments[0].userIdentifiers = arr.filter(function() {
+    return true;
+  });
+  return responseBuilder(metadata, message, destination, payload);
+};
+
+const processEvent = async (metadata, message, destination) => {
+  const { type } = message;
```

Comment:
maybe we can add one more check if `type` actually exists otherwise `.toLowerCase()` on undefined will throw some different error